### PR TITLE
Update `ActorState` impl after fvm_shared update

### PR DESCRIPTION
**Pull request by tyshko5**

### CODE REVIEW

File: fvm/Cargo.toml
Comment (Stebalien) [inline fvm/Cargo.toml:22]: Should be enabled by the "arb" feature.

File: fvm/src/state_tree.rs
Comment (Stebalien) [inline fvm/src/state_tree.rs:583]: Should be None with some probability.

Comment (tyshko5) [reply fvm/src/state_tree.rs]: Fixed accordingly

File: fvm/Cargo.toml
Comment (tyshko5) [reply fvm/Cargo.toml]: Fixed accordingly

File: fvm/src/state_tree.rs
Comment (lemmih) [inline fvm/src/state_tree.rs:579]: The `code` and `state` cid shouldn't be identical. And we don't want to duplicate the code for `Cid::arbitrary(g)`. If necessary, this PR should depend on the other PR that upgrades the `cid` dependency.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: fvm/src/state_tree.rs
Comment (tyshko5) [reply fvm/src/state_tree.rs:579]: ok https://github.com/filecoin-project/ref-fvm/pull/1107

**Review by Stebalien — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Comment by tyshko5** [comment]
@Stebalien @raulk Since this PR (https://github.com/filecoin-project/ref-fvm/pull/1107) is merged, could you, please, release new versions of the crates in this repository, after merging this PR? That would unblock some upgrades, since `cid` v0.9.0 finally got some arbitrary implementations

**Comment by tyshko5** [comment]
@Stebalien @raulk Since this PR (https://github.com/filecoin-project/ref-fvm/pull/1107) is merged, could you, please, release new versions of the crates in this repository, after merging this PR? That would unblock some upgrades, since cid v0.9.0 finally got some arbitrary implementations

**Comment by lemmih** [comment]
> @Stebalien @raulk Since this PR (#1107) is merged, could you, please, release new versions of the crates in this repository, after merging this PR? That would unblock some upgrades, since cid v0.9.0 finally got some arbitrary implementations

The v3 release is already in the works.

**Pull request merged**
